src/network/server.cc

```diff
@@ -0,0 +1,238 @@
+    packObject(child, result, list_children);
+  }
+  resp.set_ok(true);
+}
+
+void NetworkServer::createChunk(PLocalObject target_object, PLocalObject blob,
+                                network::Request &req, network::Response &resp) {
+  PLocalObject created;
+  if (target_object->type() == dbif::FILE_BLOB) {
+     created = ChunkObject::create(target_object, PLocalObject(),
+                                   req.chunk_start(), req.chunk_end(),
+                                   QString::fromStdString(req.chunk_type()),
+                                   QString::fromStdString(req.name()));
+  } else if (target_object->type() == dbif::CHUNK) {
+    created = ChunkObject::create(blob, target_object,
+                                  req.chunk_start(), req.chunk_end(),
+                                  QString::fromStdString(req.chunk_type()),
+                                  QString::fromStdString(req.name()));
+  } else {
+    resp.set_ok(false);
+    resp.set_error_msg("Bad ID provided.");
+    return;
+  }
+  network::LocalObject* result = resp.add_results();
+  packObject(created, result);
+  resp.set_ok(true);
+}
+
+void NetworkServer::deleteObject(PLocalObject target_object,
+                                 PLocalObject root_object,
+                                 network::Response &resp) {
+  if (target_object == root_object) {
+    resp.set_ok(false);
+    resp.set_error_msg("Unable to delete root.");
+    return;
+  }
+  target_object->kill();
+  resp.set_ok(true);
+}
+
+void NetworkServer::handleRequest(network::Request &req, network::Response &resp) {
+  PLocalObject target_object = root_;
+  PLocalObject blob;
+  bool found;
+
+  for (int j = 0; j < req.id_size(); j++) {
+    found = false;
+    for (auto child : target_object->children()) {
+      if (child->id() == req.id(j)) {
+        if (child->type() == dbif::FILE_BLOB) {
+          blob = child;
+        }
+        found = true;
+        target_object = child;
+        break;
+      }
+    }
+
+    if (!found) {
+      resp.set_ok(false);
+      resp.set_error_msg("Bad ID provided.");
+      return;
+    }
+  }
+
+  switch (req.type()) {
+  case 1:
+    listChildren(target_object, resp);
+    break;
+  case 2:
+    listChildren(target_object, resp, true);
+    break;
+  case 3:
+    createChunk(target_object, blob, req, resp);
+    break;
+  case 4:
+    if (target_object->type() == dbif::ROOT ||
+        target_object->type() == dbif::FILE_BLOB) {
+      resp.set_ok(false);
+      resp.set_error_msg("Unsupported object type to delete.");
+    }
+    deleteObject(target_object, root_, resp);
+    break;
+  default:
+    resp.set_ok(false);
+    resp.set_error_msg("Unknown request type.");
+    break;
+  }
+}
+
+void NetworkServer::sendResponse(QTcpSocket *client_connection,
+                                 network::Response &resp) {
+  int32_t resp_len = resp.ByteSize(), resp_len_send = qToLittleEndian(resp_len);
```

Comment:
ByteSize returned signed int so automatically stuck with it - however you are right we should use unsigned here - will change.